xbmc/platform/darwin/tvos/powermanagement/TVOSPowerSyscall.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright (C) 2012-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "TVOSPowerSyscall.h"
+
+#include "utils/log.h"
+
+IPowerSyscall* CTVOSPowerSyscall::CreateInstance()
+{
+  return new CTVOSPowerSyscall();
+}
+
+void CTVOSPowerSyscall::Register()
+{
+  IPowerSyscall::RegisterPowerSyscall(CTVOSPowerSyscall::CreateInstance);
+}
+
+bool CTVOSPowerSyscall::PumpPowerEvents(IPowerEventsCallback *callback)
+{
+  switch (m_state)
+  {
+    case SUSPENDED:
+      callback->OnSleep();
+      CLog::Log(LOGINFO, "%s: OnSleep called", __FUNCTION__);
```

Comment:
Im not sure we want this as LOGINFO ?

Sleep states with tvos are potentially very often, and just adds a lot of logging for not much benefit. Maybe better to use LOGDEBUG and only log for that.